native/src/hooks/useLocation.ts

```diff
@@ -0,0 +1,141 @@
+import Geolocation, { GeolocationError, GeolocationResponse } from '@react-native-community/geolocation'
+import { useCallback, useEffect, useState } from 'react'
+import { openSettings, RESULTS } from 'react-native-permissions'
+import SystemSetting from 'react-native-system-setting'
+
+import { checkLocationPermission, requestLocationPermission } from '../utils/LocationPermissionManager'
+
+type SuccessfulLocationState = { status: 'ready' }
+type UnavailableLocationState = {
+  status: 'unavailable'
+  message: 'noPermission' | 'notAvailable' | 'timeout' | 'loading'
+}
+
+type LocationStateType = SuccessfulLocationState | UnavailableLocationState
+
+export type LocationType = [number, number]
+
+const locationStateOnError = (error: GeolocationError): UnavailableLocationState => {
+  if (error.code === 1) {
```

Comment:
:art: only if possible: Do we have access to or can somhow use the associated constants instead of the numer, say PERMISSION_DENIED, POSITION_UNAVAILABLE, TIMEOUT?

Reply:
okay. I found out they pass the error codes with the location error. Thanks for the hint.